sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```diff
@@ -88,11 +88,18 @@ class LegacyFallbackDateFormatter(
 }
 
 object DateFormatter {
+  val defaultPattern: String = "yyyy-MM-dd"
+  val defaultLocale: Locale = Locale.US
+
   def apply(format: String, locale: Locale): DateFormatter = {
     if (SQLConf.get.legacyTimeParserEnabled) {
       new LegacyFallbackDateFormatter(format, locale)
     } else {
       new Iso8601DateFormatter(format, locale)
     }
   }
+
+  def apply(format: String): DateFormatter = apply(format, defaultLocale)
+
+  def apply(): DateFormatter = apply(defaultPattern)
```

Comment:
Both formatters seems to use thread safe implementations. You could consider just returning cached instances here.

Reply:
At the moment, both formatters are created per partition at least not per row. Do you think it makes sense to cache them?